spec/models/user_spec.rb

```diff
@@ -0,0 +1,13 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  it 'is invalid without a name' do
+    user = User.new(name: nil)
+    expect(user).to_not be_valid
+  end
+
+  it 'is valid with a non-negative integer posts counter' do
+    user = User.new(name: 'Alice', posts_counter: 0)
+    expect(user).to be_valid
+  end
```

Comment:
- [ ] Please add another test case that returns `not valid` if the `posts_counter`  has a negative value  to cover all edge cases.
```suggestion
  it 'is valid with a non-negative integer posts counter' do
    user = User.new(name: 'Alice', posts_counter: 0)
    expect(user).to be_valid
  end
  
  it 'is not valid if posts counter has a negative value' do
    user = User.new(name: 'Alice', posts_counter: -2)
    expect(user).to_not be_valid
  end
```

Or It will be much better professionally if you test the validations like this:
```ruby
describe 'validations' do
  it { should validate_presence_of(:name) }
  it { should validate_numericality_of(:posts_counter).greater_than_or_equal_to(0) }
end
```